phoenix-core/src/it/java/org/apache/phoenix/end2end/InListIT.java

```diff
@@ -537,4 +539,267 @@ public void testInListExpressionWithDesc() throws Exception {
+            assertTrue(rs.next());
+            assertEquals("000000000000500", rs.getString(1));
+        }
+    }
+
+    @Test
+    public void testInListExpressionWithRightQueryPlan() throws Exception {
+        String fullTableName = generateUniqueName();
+        String fullViewName = generateUniqueName();
+        String tenantView = generateUniqueName();
+
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.setAutoCommit(true);
+            Statement stmt = conn.createStatement();
+            stmt.execute("CREATE TABLE " + fullTableName + "(\n" + " TENANT_ID CHAR(15) NOT NULL,\n" + " KEY_PREFIX CHAR(3) NOT NULL,\n" +
+                    " CONSTRAINT PK PRIMARY KEY (\n" + " TENANT_ID," + " KEY_PREFIX" + ")) MULTI_TENANT=TRUE");
+            stmt.execute("CREATE VIEW " + fullViewName + "(\n" + " ID1 VARCHAR NOT NULL,\n" + " ID2 VARCHAR NOT NULL,\n" + " EVENT_DATE DATE, ID3 BIGINT NOT NULL \n" +
+                    " CONSTRAINT PKVIEW PRIMARY KEY\n" + " (ID1, ID2, ID3)) " +
+                    "AS SELECT * FROM " + fullTableName + " WHERE KEY_PREFIX = '0CY'");
+        }
+
+        try (Connection viewConn = DriverManager.getConnection(TENANT_SPECIFIC_URL1)) {
+            viewConn.setAutoCommit(true);
+            Statement stmt = viewConn.createStatement();
+            stmt.execute("CREATE VIEW IF NOT EXISTS " + tenantView + " AS SELECT * FROM " + fullViewName);
+
+            viewConn.createStatement().execute("UPSERT INTO " + tenantView + "(ID1, ID2, ID3) VALUES " +
+                    "('005xx000001Sv6o', '000000000000300', 1)");
+            viewConn.createStatement().execute("UPSERT INTO " + tenantView + "(ID1, ID2, ID3) VALUES " +
+                    "('005xx000001Sv6o', '000000000000400', 3)");
+            viewConn.commit();
+
+            ResultSet rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (ID1, ID2, ID3) IN " +
+                    "(('005xx000001Sv6o', '000000000000500',1)," +
+                    "('005xx000001Sv6o', '000000000000400',2)," +
+                    "('005xx000001Sv6o', '000000000000300',3))");
+
+            assertTrue(getExplainPlanString(rs).contains("DELETE SINGLE ROW"));
+
+            rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (ID2, ID1, ID3) IN " +
+                    "(('000000000000400', '005xx000001Sv6o', 1)," +
+                    "('000000000000300', '005xx000001Sv6o', 2))");
+            assertTrue(getExplainPlanString(rs).contains("DELETE SINGLE ROW"));
+
+            stmt.execute("DELETE FROM " + tenantView + " WHERE (ID3, ID2, ID1) IN " +
+                    "((1, '000000000000300', '005xx000001Sv6o'))");
+
+            rs = stmt.executeQuery("SELECT ID2 FROM " + tenantView);
+            assertTrue(rs.next());
+            assertEquals("000000000000400", rs.getString(1));
+        }
+    }
+
+    private String getExplainPlanString(ResultSet rs) throws SQLException {
```

Comment:
Doesn't a similar method already exist in PhoenixRuntime? If not consider adding to TestUtils or another class.

Reply:
removed